proto/p4/p4runtime.proto

```diff
@@ -241,14 +241,25 @@ message ActionProfileGroup {
   int32 max_size = 4;  // cannot be modified after a group has been created
 }
 
+// An index as a protobuf message. In proto3, fields cannot be optional and
+// there is no difference between an unset integer field and an integer field
+// set to 0. This is inconvenient for reading from P4 array-like structures,
+// such as indirect counters and meters. We need a way to do a wildcard read on
+// these but we cannot use a default zero index value to do so, as zero is a
+// valid index (first entry in the array). We therefore wrap the index in a
+// message.
+message Index {
+  int64 index = 1;
+}
+
 //------------------------------------------------------------------------------
 // For WriteRequest, Update.Type must be MODIFY.
 // For ReadRequest, the scope is defined as follows:
 // - All meter cells for all meters if meter_id = 0 (default).
-// - All meter cells for given meter_id if index = 0 (default).
+// - All meter cells for given meter_id if index is unset (default).
 message MeterEntry {
   uint32 meter_id = 1;
-  int64 index = 2;
+  Index index = 2;
```

Comment:
Technically, if the type changes, it can cause compatibility issues, and a new field number should be used.  We should be OK on this one because I don't see any references to either existing index outside of p4lang/PI.

Reply:
agreed, but I figured few people outside of myself must be using this, and we haven't technically released a stable P4Runtime version yet